CMakeLists.txt

```diff
@@ -65,8 +65,11 @@ include(CMakeDependentOption)
 option(BUILD_DAEMON "Build bitcoind executable." ON)
 option(BUILD_GUI "Build bitcoin-qt executable." OFF)
 option(BUILD_CLI "Build bitcoin-cli executable." ON)
-option(BUILD_TX "Build bitcoin-tx executable." ON)
-option(BUILD_UTIL "Build bitcoin-util executable." ON)
+
+option(BUILD_TESTS "Build test_bitcoin executable." ON)
+option(BUILD_TX "Build bitcoin-tx executable." ${BUILD_TESTS})
```

Comment:
We decided (on this week's call) on this approach because the functional tests depend on these binaries and silently skip tests if they're not present. It's not ideal, but we agreed it's better than the alternative.

Reply:
Ok

Reply:
A new angle on this:

Having in mind the problem that if the user compiles without e.g. `bitcoin-tx`, then some functional tests will be skipped.

Before this PR: `BUILD_TX` is `ON`, so the problem does not exist if the user uses the defaults and does not fiddle with the options. The problem arises if they explicitly set `BUILD_TX` is `OFF`.

After this PR: same as above, so this PR brings no improvement?

However, there is an undesirable side effect with this PR: if `BUILD_TESTS` is set to `OFF`, then automatically `BUILD_TX` will be turned to `OFF` as well.

Maybe it would be better if each option is kept independent. Then if `BUILD_TX` is explicitly changed from its default `ON` to `OFF`, then emit a warning during configure: "some functional tests will be skipped without bitcoin-tx".

Reply:
Two other objections to this:

* In `bitcoind` we have some options turn on/off other options. This is oftentimes complicated and difficult to asses what will happen if a given option is changed. It is hard on both ends - from the user's perspective and from developer's perspective when changing the code. I think it is better to keep it simple stupid and avoid having one option flip the value of another, which is more aligned with https://en.wikipedia.org/wiki/Principle_of_least_astonishment.

* "Full execution of functional tests" and "Whether to build unit tests executable" are unrelated and tying them together would be odd.

Reply:
I do agree with @vasild on [this](https://github.com/hebasto/bitcoin/pull/196#discussion_r1604848159):

> Maybe it would be better if each option is kept independent. Then if BUILD_TX is explicitly changed from its default ON to OFF, then emit a warning during configure: "some functional tests will be skipped without bitcoin-tx".

Reply:
I still disagree with this as I'd rather err on the side of "whoops, I built an extra binary but all the tests got run".